.github/workflows/docs.yml

```
@@ -0,0 +1,32 @@
+name: github pages
+permissions:
+  contents: write
+on:
+  push:
+    branches:
+      - main  # Set a branch to deploy
+      - doc_deploy
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true  # Fetch Hugo themes (true OR recursive)
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
```

Let's pin the actions to the specific commits:
```suggestion
        uses: peaceiris/actions-hugo@c03b5dbed22245418539b65eb9a3b1d5fdd9a0a6 # v2.5.0
```